parsel/selector.py

```diff
@@ -223,6 +237,44 @@ def __getstate__(self):
     def _get_root(self, text, base_url=None):
         return create_root_node(text, self._parser, base_url=base_url)
 
+    def jmespath(self, query, **kwargs):
+        """
+        Call the ``.jmespath()`` method for each element in this list and return
+        their results flattened as another :class:`SelectorList`.
+
+        ``query`` is the same argument as the one in :meth:`Selector.jmespath`
+
+        Requires : jmespath(https://github.com/jmespath/jmespath)
+        """
+        try:
+            if self.json is not None:
+                datas = self.json
+            elif self._text is not None:
+                datas = json.loads(self._text)
+            else:
+                if isinstance(self.root, six.string_types):
+                    datas = json.loads(self.root)
+                else:
+                    datas = json.loads(self.root.text)
+        except jmespath.exceptions.JMESPathError as exc:
```

Comment:
How can this exception happen, if `jmespath` is not used within the `try`?

On the other hand, it looks like there is a good chance of a ValueError happening, if any of the arguments to `json.loads` are not valid JSON. It may make sense to let such an exception raise normally, though.